bridge/engine/EngineMixer.cpp

```diff
@@ -3834,15 +3838,57 @@ void EngineMixer::checkIfRateControlIsNeeded(const uint64_t timestamp)
     _probingVideoStreams = enableBEProbing;
 }
 
-void EngineMixer::runTransportTicks(const uint64_t timestamp)
+template <typename TStream>
+void removeIdleStreams(concurrency::MpmcHashmap32<size_t, TStream*>& streams,
+    std::function<void(TStream*)> removeMethod,
+    const uint64_t timestamp)
 {
-    for (auto videoIt : _engineVideoStreams)
+    for (const auto& it : streams)
     {
-        auto& videoStream = *videoIt.second;
-        videoStream.transport.runTick(timestamp);
+        auto stream = it.second;
+        if (!stream->idleTimeoutSeconds)
+        {
+            continue;
+        }
+
+        if (utils::Time::diffLT(stream->createdAt, timestamp, stream->idleTimeoutSeconds * utils::Time::sec))
+        {
+            continue;
+        }
+
+        const auto lastReceivedTimestamp = stream->transport.getLastReceivedPacketTimestamp();
+        if (utils::Time::diffGE(lastReceivedTimestamp, timestamp, stream->idleTimeoutSeconds * utils::Time::sec))
+        {
+            logger::info("Removing idle stream for endpoint %zu, last received packet timestamp: %zu, current: %zu",
+                "EngineMixer",
+                stream->endpointIdHash,
+                lastReceivedTimestamp,
+                timestamp);
+            removeMethod(stream);
+        }
+    }
+}
+
+#define REMOVE_IDLE_STREAMS(type)                                                                                      \
+    ::bridge::removeIdleStreams<Engine##type##Stream>(_engine##type##Streams,                                          \
+        std::bind(&EngineMixer::remove##type##Stream, this, std::placeholders::_1),                                    \
```

Comment:
`std::bind` most probably will create a dynamic memory which is something we don't allow on RT thread. There is a recommendation to std::function to have a min storage to have at least space to a pointer without a dynamic allocation but even that is not ensured by standard although we now most used compilers will not allocate if you capture 8 bytes on a lambda expression.

But as far as I know `std::bind` due to the complexity of implementation always use dynamic memory.

Reply:
Removed lambda/binding and instead overloaded removeStream(...) in mixer.